connectors/es/sink.py

```diff
@@ -171,6 +180,43 @@ async def _bulk_api_call():
 
         return res
 
+    async def _log_bulk_operations(self, res):
+        for item in res.get("items", []):
+            if "index" in item:
+                action_item = "index"
+            elif "delete" in item:
+                action_item = "delete"
+            elif "create" in item:
+                action_item = "create"
+            elif "update" in item:
+                action_item = "update"
+            else:
+                # Should only happen, if the _bulk API changes
+                # Unlikely, but as this functionality could be used for audits we want to detect changes fast
+                self._logger.error(
+                    f"Unknown action item returned from _bulk API for item {item}"
+                )
+                return
+
+            doc_id = item[action_item].get("_id")
+            if doc_id is None:
+                # Should only happen, if the _bulk API changes
+                # Unlikely, but as this functionality could be used for audits we want to detect changes fast
+                self._logger.error(f"Could not retrieve '_id' for item {item}")
+                return
+
+            result = item[action_item].get("result")
+            operation_failed = result not in SUCCESSFUL_RESULTS
```

Comment:
According to the [doc](https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-bulk.html#bulk-api-response-body):

>result
>(string) Result of the operation. Successful values are created, deleted, and updated.
>
>This parameter is only returned for successful operations.

https://github.com/elastic/connectors/blob/f4d65e252bae32d228f0e776e517977f5d2ba5ab/connectors/es/sink.py#L176-L178

Reply:
I think there's a difference between a successful operation and a successful value. A successful operation could also be a operation, where the result is `not_found` (which is a non-successful value), which we might look at as the operation failed as we didn't find an id the connector expected to be there for some reason. WDYT?